### DISCUSSION TITLE
Can different argocd applications share namespace without problem?

### DISCUSSION THREAD

**Discussion by byvinx** [argoproj/argo-cd#10790] [General]
My git repository has a "helm" folder useful for a subchart deploy pattern and a "yaml" folder where instead the manifest k8s are simply listed, and are managed on the operation side, eg: monitor, service account .. In fact both helm and the manifest are used to run the application, but are generated by different groups with different technologies (helm package on artifactory by pipeline and simple manifest directory manually) .Now in a single app on argo cd it is not possible to have both technologies, that is the helm path in one source and the yaml path in another. So I created two argocd apps, one pointing to the repo on the path with the helm dependencies, the other pointing to the repo on the yaml path, but on the same namespace.  Everything works but there might be conflicts or problems with pruning or other features?

**Comment by jgwest** [comment]
Without speaking to your specific use case, AFAIK it is fine to have two Argo CD applications targeting the same namespace, as long as:
1) There exist no resources that are shared between the 2 Applications. (Argo CD will usually loudly warn about this)
2) The label which Argo CD applies to resources that it deploys is not removed. The label that is applied indicates which Argo CD Application a particular resource belongs to, allowing it to tell the two apart.

**Reply by byvinx** [reply]
[Here](https://github.com/argoproj/argo-cd/pull/10432) seems that the possibility of having two separate sources will be released soon, even with Multiple Value files for Helm Source. This simplifies things, managing in a single app manifest what I previously had to do with two manifest apps sharing the namespace.
